app/controllers/courses_controller.rb

```diff
@@ -38,6 +38,7 @@ def show
       redirect_to info_course_path(course)
       return
     end
+    @enrollment.reminder.update_visit_time
```

Comment:
why an extra level of indirection? just update the enrollment directly.

also, you can use the `touch` method to update timestamp.